src/view/com/composer/videos/state.ts

```diff
@@ -0,0 +1,49 @@
+import {useState} from 'react'
+import {ImagePickerAsset} from 'expo-image-picker'
+import {msg} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+import {useMutation} from '@tanstack/react-query'
+
+import {compressVideo} from '#/lib/media/video/compress'
+
+export function useVideoState({setError}: {setError: (error: string) => void}) {
+  const {_} = useLingui()
+  const [progress, setProgress] = useState(0)
+
+  const {mutate, data, isPending, isError, reset, variables} = useMutation({
+    mutationFn: async (asset: ImagePickerAsset) => {
+      if (asset.fileSize) {
+        console.log(
+          'uncompressed size',
+          (asset.fileSize / 1024 / 1024).toFixed(2) + 'mb',
+        )
+      }
+      const compressed = await compressVideo(asset.uri, {
+        onProgress: num => setProgress(trunc2dp(num)),
+      })
+
+      return compressed
+    },
+    onError: error => {
+      console.error('error', error)
```

Comment:
log

Reply:
can this error for any reason other than the throw in `compressVideo`? if so would be nice to use `logger.error()` for things that are _not_ the `greater than 100mb` error just so we don't spam sentry. otherwise i think just toast is fine.

Reply:
I'll just show toast for now